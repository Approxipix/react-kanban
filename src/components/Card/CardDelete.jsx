import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { deleteCard } from '../../redux/cardReducer/actions/actions';
import { Text, SubmitButton } from '../BaseComponent';
import { history } from "../../redux/store";

class CardDelete extends Component {
  deleteList = () => {
    const { listId, cardId, currentBoardID } = this.props;
    this.props.actions.deleteCard({
      listId: listId,
      cardId: cardId,
    });
    history.push(`/b/${currentBoardID}`)
  };

  render() {
    return (
      <>
        <Text>
          All actions will be removed from the activity feed and you wonâ€™t be able to re-open the card.
          There is no undo.
        </Text>
        <SubmitButton color={'danger'} onClick={() => this.deleteList()}>
          Delete card
        </SubmitButton>
      </>
    );
  }
}

function mapStateToProps(state) {
  return {
    currentBoardID: state.rootReducer.currentBoardID,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteCard: deleteCard,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(CardDelete);
