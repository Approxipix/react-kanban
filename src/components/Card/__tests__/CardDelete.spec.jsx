import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import CardDelete from '../CardDelete';

const mockStore = configureStore([]);

describe('<CardDelete>', () => {
  let store;
  let component;
  let mockDeleteCard = jest.fn();

  beforeEach(() => {
    const props = {
      listId: 'ListID',
      cardId: 'CardID',
    };
    store = mockStore({
      rootReducer: {
        currentBoardID: 'BoardID'
      }
    });
    store.dispatch = mockDeleteCard;

    const wrapper = shallow(<CardDelete store={store} {...props} />);
    component = wrapper.find('CardDelete').dive();
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should dispatch an action after click on button", () => {
    const expectedAction = {
      type: 'DELETE_CARD',
      payload: {
        listId: 'ListID',
        cardId: 'CardID',
      }
    };

    component.find('SubmitButton').simulate('click');
    expect(mockDeleteCard).toHaveBeenCalledTimes(1);
    expect(mockDeleteCard).toHaveBeenCalledWith(expectedAction);
  });
});
