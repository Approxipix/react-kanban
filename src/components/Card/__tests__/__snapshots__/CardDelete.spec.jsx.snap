// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDelete> should render with given state from Redux store 1`] = `
<Fragment>
  <styled.p>
    All actions will be removed from the activity feed and you wonâ€™t be able to re-open the card. There is no undo.
  </styled.p>
  <SubmitButton
    color="danger"
    onClick={[Function]}
  >
    Delete card
  </SubmitButton>
</Fragment>
`;
